sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/unresolved.scala

```diff
@@ -342,7 +342,8 @@ case class UnresolvedFunction(
     isDistinct: Boolean,
     filter: Option[Expression] = None,
     ignoreNulls: Boolean = false,
-    orderingWithinGroup: Seq[SortOrder] = Seq.empty)
+    orderingWithinGroup: Seq[SortOrder] = Seq.empty,
+    isInternal: Boolean = false)
```

Comment:
`UnresolvedFunction` is an internal class, but I'm wondering if it's safer to add `ExtendedUnresolvedFunction` so that we can keep backward compatibility for `UnresolvedFunction` which might be used by custom catalyst rules?

Reply:
I think it is fine for two reasons:
1. It should be very rare to add an internal function. As a 3p extension developer you generally want the SQL Dataframe surface to match.
2. A 3p developer should not really be creating these UnresolvedFunctions. This should be done using the Column API, and indirectly through the Spark API.